src/DotNetBoilerplate.Api/BookStores/GetBookStoresEndpoint.cs

```diff
@@ -0,0 +1,43 @@
+﻿using DotNetBoilerplate.Application.BookStores.Get;
+using DotNetBoilerplate.Application.BookStores.Browse;
+using DotNetBoilerplate.Application.BookStores.DTO;
+using DotNetBoilerplate.Shared.Abstractions.Queries;
+using Microsoft.AspNetCore.Http.HttpResults;
+using Microsoft.AspNetCore.Mvc;
+
+namespace DotNetBoilerplate.Api.BookStores;
+
+public class GetBookStoresEndpoint : IEndpoint
+{
+    public static void Map(IEndpointRouteBuilder app)
+    {
+        app.MapGet("", Handle)
+            .WithSummary("Browse book store with optional Id parameter");
+    }
+
+    private static async Task<Results<Ok<IEnumerable<BookStoreDto>>, Ok<BookStoreDto>, NotFound>> Handle(
+        [FromQuery] Guid? id,
+        [FromServices] IQueryDispatcher queryDispatcher,
+        CancellationToken ct
+    )
+    {
+        if (id.HasValue)
+        {
+            var query = new GetBookStoreByIdQuery(id.Value);
```

Comment:
Nie musimy dzielić tego na GetBookStoreByIdQuery i BrowseBookStoresQuery tutaj. Może się wczoraj źle zrozumieliśmy. Chodzi mi o to że jak mamy endpoint gdzie chcemy pobrać wszystkie zasoby to może on wyglądać
/book-stores , Jak chcemy w nim też coś filtrować po id to możemy ten id dodać do query params np, /book-stores?bookStoreId=1 . To nie znaczy że powinniśmy robić 2 osobne queries i handlers. Ten id z parametru powinien być nullowalny i dopiero na samym poziomie query handlera sprawdzamy czy ten parametr został podany czy nie i jeśli odpowiedź brzmi tak to wtedy filtrujemy zasób po tym id